File: go/pkg/models/character_location.go
package models

import (
  "time"

  "github.com/google/uuid"
)

type CharacterLocation struct {
  ID          uuid.UUID `json:"id" gorm:"type:uuid;default:uuid_generate_v4()"`
  CreatedAt   time.Time `json:"createdAt"`
  UpdatedAt   time.Time `json:"updatedAt"`
  CharacterID uuid.UUID `json:"characterId" gorm:"type:uuid"`
  Latitude    float64   `json:"latitude"`
  Longitude   float64   `json:"longitude"`
}

func (c *CharacterLocation) TableName() string {
  return "character_locations"
}

File: dart/unclaimed_streets/lib/models/character_location.dart
class CharacterLocation {
  final String id;
  final String characterId;
  final double latitude;
  final double longitude;

  const CharacterLocation({
    required this.id,
    required this.characterId,
    required this.latitude,
    required this.longitude,
  });

  factory CharacterLocation.fromJson(Map<String, dynamic> json) {
    return CharacterLocation(
      id: json['id'] as String? ?? '',
      characterId: json['characterId'] as String? ?? '',
      latitude: (json['latitude'] as num?)?.toDouble() ?? 0,
      longitude: (json['longitude'] as num?)?.toDouble() ?? 0,
    );
  }
}
